internal/settings/providers/allinkl/provider.go

```diff
@@ -0,0 +1,172 @@
+	case len(p.password) == 0:
+		return errors.ErrEmptyPassword
+	case p.host == "*":
+		return errors.ErrHostWildcard
+	}
+	return nil
+}
+
+func (p *provider) String() string {
+	return utils.ToString(p.domain, p.host, constants.AllInkl, p.ipVersion)
+}
+
+func (p *provider) Domain() string {
+	return p.domain
+}
+
+func (p *provider) Host() string {
+	return p.host
+}
+
+func (p *provider) IPVersion() ipversion.IPVersion {
+	return p.ipVersion
+}
+
+func (p *provider) Proxied() bool {
+	return false
+}
+
+func (p *provider) BuildDomainName() string {
+	return utils.BuildDomainName(p.host, p.domain)
+}
+
+func (p *provider) HTML() models.HTMLRow {
+	return models.HTMLRow{
+		Domain:    models.HTML(fmt.Sprintf("<a href=\"http://%s\">%s</a>", p.BuildDomainName(), p.BuildDomainName())),
+		Host:      models.HTML(p.Host()),
+		Provider:  "<a href=\"https://all-inkl.com/\">ALL-INKL.com</a>",
+		IPVersion: models.HTML(p.ipVersion.String()),
+	}
+}
+
+func (p *provider) Update(ctx context.Context, client *http.Client, ip net.IP) (newIP net.IP, err error) {
+	u := url.URL{
+		Scheme: "https",
+		Host:   "dyndns.kasserver.com",
+		Path:   "/",
+		User:   url.UserPassword(p.username, p.password),
+	}
+	values := url.Values{}
+	values.Set("host", utils.BuildURLQueryHostname(p.host, p.domain))
+	if !p.useProviderIP {
+		if ip.To4() == nil { // ipv6
+			values.Set("myip", ip.String())
+		} else {
+			values.Set("myip", ip.String())
+		}
+	}
+	u.RawQuery = values.Encode()
+
+	request, err := http.NewRequestWithContext(ctx, http.MethodGet, u.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+	headers.SetUserAgent(request)
+
+	response, err := client.Do(request)
+	if err != nil {
+		return nil, err
+	}
+	defer response.Body.Close()
+
+	b, err := io.ReadAll(response.Body)
+	if err != nil {
+		return nil, fmt.Errorf("%w: %s", errors.ErrUnmarshalResponse, err)
+	}
+	s := string(b)
+
+
+	if response.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("%w: %d: %s", errors.ErrBadHTTPStatus, response.StatusCode, s)
+	}
+
+	switch s {
+	case "":
+		return nil, errors.ErrNoResultReceived
+	case constants.Nineoneone:
+		return nil, errors.ErrDNSServerSide
+	case constants.Abuse:
+		return nil, errors.ErrAbuse
+	case "!donator":
+		return nil, errors.ErrFeatureUnavailable
+	case constants.Badagent:
+		return nil, errors.ErrBannedUserAgent
+	case constants.Badauth:
+		return nil, errors.ErrAuth
+	case constants.Nohost:
+		return nil, errors.ErrHostnameNotExists
+	}
+	if strings.Contains(s, "nochg") || strings.Contains(s, "good") {
+		ips := verification.NewVerifier().SearchIPv4(s)
+		if ips == nil {
+			return nil, errors.ErrNoResultReceived
+		}
+		newIP = net.ParseIP(ips[0])
+		if newIP == nil {
+			return nil, fmt.Errorf("%w: %s", errors.ErrIPReceivedMalformed, ips[0])
+		}
+		if !p.useProviderIP && !ip.Equal(newIP) {
+			return nil, fmt.Errorf("%w: %s", errors.ErrIPReceivedMismatch, newIP.String())
+		}
+		return newIP, nil
+	}
```

Comment:
Please (to get the error `errors.ErrNoIPInResponse`)

```sh
git remote add upstream https://github.com/qdm12/ddns-updater.git
git fetch upstream
git rebase upstream/master
```

Then change this code to

```suggestion
	if !strings.Contains(s, "nochg") && !strings.Contains(s, "good") {
		return nil, fmt.Errorf("%w: %s", errors.ErrUnknownResponse, s)
	}

	var ips []string
	verifier := verification.NewVerifier()
	if ip.To4() != nil {
		ips = verifier.SearchIPv4(s)
	} else {
		ips = verifier.SearchIPv6(s)
	}

	if len(ips) == 0 {
		return nil, errors.ErrNoIPInResponse
	}

	newIP = net.ParseIP(ips[0])
	if newIP == nil {
		return nil, fmt.Errorf("%w: %s", errors.ErrIPReceivedMalformed, ips[0])
	}
	if !p.useProviderIP && !ip.Equal(newIP) {
		return nil, fmt.Errorf("%w: %s", errors.ErrIPReceivedMismatch, newIP.String())
	}
	return newIP, nil
```

And finally `git push origin master --force`

Thanks!